girder/models/user.py

```diff
@@ -244,26 +249,43 @@ def createUser(self, login, password, firstName, lastName, email,
             'groupInvites': []
         }
 
-        self.setPassword(user, password)
+        self.setPassword(user, password, save=False)
+
+        self.setPublic(user, public, save=False)
+
+        # An id is required in order to grant the user access on itself, but
+        # since the user isn't fully created yet, it should not be saved
+        user['_id'] = ObjectId()
```

Comment:
If you think it's important, though, I think the best idea would be to add logic to `AccessControlledModel.hasAccess` that special-cases for users and groups at runtime and automatically returns True if the user is checking their own document, or for a group to which they belong. It might be cleanest to override `hasAccess` in the group and user models.